File: NetBet/ApiControllers/PlayerController.fs
namespace NetBet.Controllers

open System
open System.Collections.Generic
open System.Linq
open Microsoft.AspNetCore.Mvc
open DbTypes
open DtoTypes

[<Route("api/[controller]")>]
[<ApiController>]
type PlayerController () =
    inherit ControllerBase()
    
    [<HttpGet>]
    member __.Get() =
        let players = SeasonService.getAllPlayers()
        ActionResult<Player[]>(players)

    [<HttpGet("{seasonid}")>]
    member __.GetPlayersForSeason(seasonid: int) =
        let sp = SeasonService.getPlayersForSeason seasonid
        ActionResult<SeasonPlayer[]>(sp)
        
    [<HttpGet>]
    member __.GetEmpty() =
        let emptyPlayer = 
            { ID            = 0
              Name          = ""}
        ActionResult<Player>(emptyPlayer)
        
    [<HttpPost>]
    member __.CreateOrUpdate([<FromBody>] p: Player) =
        SeasonService.createOrUpdatePlayer p |> ignore
        
    [<HttpDelete("{id}")>]
    member __.Delete(id:int) =
        SeasonService.deletePlayer id

File: NetBet.WebApp/ApiControllers/PlayerController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using static DbTypes;

namespace NetBet.WebApp.ApiControllers
{
    [Route("api/[controller]/[action]")]
    [ApiController]
    public class PlayerController : ControllerBase
    {
        [HttpGet]
        public Player[] Get()
        {
            return SeasonService.getAllPlayers();
        }
        
        [HttpGet("{seasonid}")]
        public SeasonPlayer[] GetPlayersForSeason(int seasonid)
        {
            return SeasonService.getPlayersForSeason(seasonid);
        }
        
        [HttpPost]
        public void CreateOrUpdate([FromBody] Player p)
        {
            SeasonService.createOrUpdatePlayer(p);
        }
        
        
        [HttpDelete("{id}")]
        public void Delete(int id)
        {
            SeasonService.deletePlayer(id);
        }
    }
}
